bench/bench-case-folding.cpp

```diff
@@ -0,0 +1,21 @@
+#include "folding.hh"
+#include <benchmark/benchmark.h>
+
+static std::u32string a = U"HOMEBREW_NO_AUTO_UPDATE. Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).";
+
+static void applyFolding( benchmark::State & state )
+{
+  for ( auto _ : state ) {
+    Folding::apply( a );
+  }
+}
+BENCHMARK( applyFolding );
+
+
+static void applyFolding2( benchmark::State & state )
+{
+  for ( auto _ : state ) {
+    Folding::apply2( a );
+  }
+}
```

Comment:
those two functions are not the same.
Qt's caseFolding only support Simple Folding and not Complex Folding .   the implementation of qt casefolding is not complete .

Reply:
https://bugreports.qt.io/browse/QTBUG-112748